test/e2e/storage/testsuites/fsgroupchangepolicy.go

```diff
@@ -0,0 +1,282 @@
+}
+
+func (s *fsGroupChangePolicyTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	type local struct {
+		config        *PerTestConfig
+		driverCleanup func()
+
+		cs        clientset.Interface
+		ns        *v1.Namespace
+		driver    TestDriver
+		resources []*VolumeResource
+	}
+	var l local
+	ginkgo.BeforeEach(func() {
+		dInfo := driver.GetDriverInfo()
+		if !dInfo.Capabilities[CapFsGroup] {
+			e2eskipper.Skipf("Driver %q does not support FsGroup - skipping", dInfo.Name)
+		}
+
+		if pattern.VolMode == v1.PersistentVolumeBlock {
+			e2eskipper.Skipf("Test does not support non-filesystem volume mode - skipping")
+		}
+
+		if pattern.VolType != testpatterns.DynamicPV {
+			e2eskipper.Skipf("Suite %q does not support %v", s.tsInfo.Name, pattern.VolType)
+		}
+
+		_, ok := driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.VolType)
+		}
+
+		if framework.NodeOSDistroIs("windows") {
+			e2eskipper.Skipf("FSGroupPolicy is only applied on linux nodes -- skipping")
+		}
+	})
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("fsgroupchangepolicy")
+
+	init := func() {
+		l = local{}
+		l.ns = f.Namespace
+		l.cs = f.ClientSet
+		l.driver = driver
+		l.config, l.driverCleanup = driver.PrepareTest(f)
+	}
+
+	cleanup := func() {
+		var errs []error
+		for _, resource := range l.resources {
+			errs = append(errs, resource.CleanupResource())
+		}
+
+		errs = append(errs, tryFunc(l.driverCleanup))
+		l.driverCleanup = nil
+		framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup resource")
+	}
+
+	ginkgo.It("pod fsgroup change policy (Always) for dynamic provisioned volume", func() {
+		init()
+		defer cleanup()
+
+		testVolumeSizeRange := s.GetTestSuiteInfo().SupportedSizeRange
+		resource := CreateVolumeResource(l.driver, l.config, pattern, testVolumeSizeRange)
+		l.resources = append(l.resources, resource)
+		TestPodFsGroupChangePolicyAlways(l.config.Framework, l.cs, l.ns.Name, l.config.ClientNodeSelection, resource.Pvc)
+	})
+
+	ginkgo.It("pod fsgroup change policy (OnRootMismatch) for dynamic provisioned volume", func() {
+		init()
+		defer cleanup()
+
+		testVolumeSizeRange := s.GetTestSuiteInfo().SupportedSizeRange
+		resource := CreateVolumeResource(l.driver, l.config, pattern, testVolumeSizeRange)
+		l.resources = append(l.resources, resource)
+		TestPodFsGroupChangePolicyOnRootMismatch(l.config.Framework, l.cs, l.ns.Name, l.config.ClientNodeSelection, resource.Pvc)
+	})
+}
+
+// TestPodFsGroupChangePolicyAlways tests 'Always' policy.
+func TestPodFsGroupChangePolicyAlways(f *framework.Framework, cs clientset.Interface,
```

Comment:
minor nit: can this function not inferece namespace and clientset from framework object?

Reply:
Ack we can.